inbox/crispin.py

```diff
@@ -376,7 +376,14 @@ def _fetch_folder_list(self):
         IMAPClient parses this response into a list of
         (flags, delimiter, name) tuples.
         """
-        return self.conn.list_folders()
+
+        # As discovered in the wild list_folders() can return None as name,
+        # we cannot handle those folders anyway so just filter them out.
+        return [
+            (flags, delimiter, name)
+            for flags, delimiter, name in self.conn.list_folders()
+            if name
```

Comment:
Is a folder valid when it has an empty string as its name? Maybe we should check `if name is None` instead.

I agree about skipping invalid folders 👍

Reply:
I've actually searched our production data and there are no empty string folder names, but I don't mind checking `None` explicitely.